python/rmm/tests/test_rmm.py

```diff
@@ -942,3 +943,73 @@ def test_rmm_device_buffer_copy(cuda_ary, make_copy):
     result = db_copy.copy_to_host()
 
     np.testing.assert_equal(expected, result)
+
+
+@pytest.mark.parametrize("level", rmm.logging_level)
+def test_valid_logging_level(level):
+    warnings.filterwarnings(
+        "ignore", message="RMM will not log warning level TRACE."
+    )
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** ‏These calls (and in the test below) are side-effectful for the lifetime of the program, rather than just the duration of the scope they were called in. I think we want to only filter these warnings out in this scope. For which we want to use the `warnings.catch_warnings()` context manager:

```python
def test_valid_logging_level(...):
    with warnings.catch_warnings():
        warnings.filterwarnings("ignore", message=...)
        warnings.filterwarnings("ignore", message=...)
        ...
        # test body here
```

Same comment applies to the test below as well.